code/weapon/weapons.cpp

```diff
@@ -2353,6 +2353,33 @@ int parse_weapon(int subtype, bool replace, const char *filename)
 		}
 	}
 
+	while (optional_string("$Conditional Impact Effect:")) {
+		int armor_index;
+		ConditionalImpact ci;
+		ci.min_health_threshold = 0.0;
+		ci.max_health_threshold = 1.0;
+		ci.min_angle_threshold = 0.0;
+		ci.max_angle_threshold = 180.0;
+		ci.dinky = false;
+		required_string("+Armor Type:");
+			stuff_string(fname, F_NAME, NAME_LENGTH);
+		armor_index = armor_type_get_idx(fname);
+		required_string("+Effect Name:");
+			ci.effect = particle::util::parseEffect(wip->name);
+		parse_optional_float_into("+Min Health Threshold:", &ci.min_health_threshold);
+		parse_optional_float_into("+Max Health Threshold:", &ci.max_health_threshold);
+		parse_optional_float_into("+Min Angle Threshold:", &ci.min_angle_threshold);
+		parse_optional_float_into("+Max Angle Threshold:", &ci.max_angle_threshold);
+		parse_optional_bool_into("+Dinky:", &ci.dinky);
+		SCP_vector<ConditionalImpact> ci_vec;
+		if (wip->conditional_impacts.count(armor_index) == 1) {
```

Comment:
I was going to balk at `count() == 1` being used to determine if a key exists but apparently that's the typical way of doing this!

Reply:
As far as I can tell it's the _only_ way of doing this. I tried other options!